### ISSUE TITLE
create disposable store for accessible view render

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#186537]
It doesn't matter in this case, but note that you collect all the keyListener disposables in the store of the AccessibleView class (even if you dispose them in `onHide`, but the disposable is still in the disposableStore). This can lead to memory leaks.

I can really recommend to compose components according to their life times.
I think here the lifetime of the rendered content is owned by the caller of `_render` (the caller is supposed to call `.dispose()` on the result).
So the disposable returned by `onKeyUp` should be owned by whatever you return in `_render`.
I suggest to create a DisposableStore in _render, push the key listener to it once the text model resolves and return the disposable store at the end of _render (also add the `provider.onClose()` to the store).
Also, note that the _getTextModel promise can resolve after the caller called dispose.

_Originally posted by @hediet in https://github.com/microsoft/vscode/pull/186504#discussion_r1245527773_
